playbook/app/pb_kits/playbook/pb_advanced_table/advanced_table.test.jsx

```diff
@@ -53,4 +132,214 @@ test("generated scaffold test", () => {
+      <AdvancedTable.Body subRowHeaders={subRowHeaders}/>
+    </AdvancedTable>
+    );
+
+  const kit = screen.getByTestId(testId);
+
+  const rowButton = kit.querySelector(".gray-icon.expand-toggle-icon")
+  rowButton.click()
+  
+  const subRowHeader = kit.querySelector(".custom-row.bg-silver")
+  expect(subRowHeader).toBeInTheDocument()
+});
+
+test("expandControl prop works as expected", () => {
+  render (<AdvancedTableExpandControl/>)
+
+  const kit = screen.getByTestId(testId);
+  const subRow = kit.querySelector(".bg-white.depth-sub-row-1")
+  expect(subRow).toBeInTheDocument()
+})
+
+test("tableOptions prop functions as expected", () => {
+  render(
+    <AdvancedTable
+        columnDefinitions={columnDefinitions}
+        data={{ testid: testId }}
+        tableData={MOCK_DATA}
+        tableOptions={tableOptions}
+    />
+  );
+
+  const kit = screen.getByTestId(testId);
+  const row1 = kit.getElementsByTagName('tr')[1]
+  
+  expect(row1.id).toBe("1-1-0-row")
+});
+
+test("tableProps prop functions as expected", () => {
+  render(
+    <AdvancedTable
+        columnDefinitions={columnDefinitions}
+        data={{ testid: testId }}
+        tableData={MOCK_DATA}
+        tableProps={tableProps}
+    />
+  );
+
+  const kit = screen.getByTestId(testId);
+  const table = kit.querySelector('table')
+  expect(table).toHaveClass("pb_table table-sm table-responsive-none data_table sticky-header ns_tabular")
+});
+
+test("enableExpansionIcon changes icon", () => {
+  render(
+    <AdvancedTable
+        columnDefinitions={columnDefinitions}
+        data={{ testid: testId }}
+        tableData={MOCK_DATA}
+        tableProps={tableProps}
+        toggleExpansionIcon= "chevron-up"
+    />
+  );
+
+  const kit = screen.getByTestId(testId);
+  const tableHead = kit.querySelector('table')
+  const toggleIcon= tableHead.querySelector(".pb_icon_kit")
+  expect(toggleIcon).toHaveClass("fa-chevron-up")
+});
+
+test("sortIcon changes icon", () => {
+  render(
+    <AdvancedTable
+        columnDefinitions={columnDefinitions}
+        data={{ testid: testId }}
+        tableData={MOCK_DATA}
+        tableProps={tableProps}
+    >
+      <AdvancedTable.Header 
+          enableSorting
+          sortIcon= {["chevron-up","chevron-down"]}
+      />
+      <AdvancedTable.Body />
+    </AdvancedTable>
+  );
+
+  const kit = screen.getByTestId(testId);
+  const sortIcon = kit.querySelector('.sort-button-icon')
+  const icon= sortIcon.querySelector(".pb_icon_kit")
+  expect(icon).toHaveClass("fa-chevron-down")
+});
+
+test("Sort icon renders with enableSorting + sortControl works as expected", () => {
+  render (<AdvancedTableSortControl/>)
+
+  const kit = screen.getByTestId(testId);
+  const sortIcon = kit.querySelector(".sort-button-icon")
+  expect(sortIcon).toBeInTheDocument()
+
+  const row1 = kit.getElementsByTagName('tr')[1]
+  
+  expect(row1.id).toBe("1-1-0-row")
+})
+
+// test("sort button exists and sorts column data", () => {
```

Comment:
This test is correct, but it is commented out because there is a bug on the sort functionality that is being corrected on PBNTR-178 [here](https://github.com/powerhome/playbook/pull/3102/files#diff-72fb058672927e1a48a2a4cc6a8fed0e7e9bb8ebd45ca194fd6b90ae4874ee20R165). The test will work correctly once that PR is merged, but will fail without it. So I'm writing the test and leaving it commented out for now, once PBNTR-178 is merged, we can merge master and comment this test back in before merging.

@jasperfurniss